hikari/impl/rest.py

```diff
@@ -1446,6 +1451,9 @@ def _build_message_payload(  # noqa: C901- Function too complex
         elif attachments:
             final_attachments.extend(attachments)
 
+        if len(final_attachments) != 1 and not undefined.any_undefined(waveform, duration):
+            raise ValueError("You can only have one attachment when you specify waveform and duration!")
```

Comment:
If we only allow specifying one attachment, then we do not need to check this again. And again, we could just leave this up to Discord